### ISSUE TITLE
Failed to detect Deno on Linux

### ISSUE THREAD

**Issue by David-Else** [justjavac/vscode-deno#59]
**Describe the bug**

I get this error after installing the plugin:
`Failed to detect Deno.
You can use one-line commands to install Deno.`

but Deno is installed and in my $PATH:
```
$ deno info
DENO_DIR location: "/home/xxx/.cache/deno"
Remote modules cache: "/home/xxx/.cache/deno/deps"
TypeScript compiler cache: "/home/xxx/.cache/deno/gen"
```

```
# .bash_profile
PATH=$PATH:$HOME/.deno/bin
```
```
$whereis deno
deno: /home/xxx/.deno/bin/deno
```
### Versions

```
Name: Deno
Id: justjavac.vscode-deno
Description: Deno support for VSCode
Version: 1.3.3
Publisher: justjavac
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=justjavac.vscode-deno
```
```
VS Code 1.39.2 - Centos8 Linux
```

**Comment by justjavac** [comment]
try 1.4.0

**Comment by barthuijgen** [comment]
Having the same issue using

```
Name: Deno
Id: justjavac.vscode-deno
Description: Deno support for VSCode
Version: 1.4.0
Publisher: justjavac
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=justjavac.vscode-deno
```

deno v0.26.0 info
```
DENO_DIR location: "/Users/barthuijgen/Library/Caches/deno"
Remote modules cache: "/Users/barthuijgen/Library/Caches/deno/deps"
TypeScript compiler cache: "/Users/barthuijgen/Library/Caches/deno/gen"
```

It seems `~/.deno` isn't being used anymore, not sure since which version. I've updated my paths to reflect new location of deno which is 

```
> which deno
/Users/barthuijgen/.local/bin/deno
```

**Comment by David-Else** [comment]
@justjavac 1.4.0 is not in market place, it is 1.3.3

**Comment by justjavac** [comment]
fixed in https://github.com/justjavac/vscode-deno/commit/2565d1f6d25e88f61ed34fabdf7c4b55c4bbf15d

and the latest version is v1.4.0

https://marketplace.visualstudio.com/items?itemName=justjavac.vscode-deno

![](https://user-images.githubusercontent.com/359395/70618768-9ae5ce00-1c4e-11ea-838d-766be28d7750.png)

**Comment by barthuijgen** [comment]
I was already using `1.4.0` as said in my post before. Still getting this same issue, how can i find out what is causing it?

**Comment by justjavac** [comment]
I don't have a linux system for debugging. But this week I will release 1.5.0 version to add more debugging info

**Comment by David-Else** [comment]
I have tried totally uninstalling and re-installing, still it uses version `1.3.3`... maybe there is a bug in your version number strings or something like that?

![Screenshot from 2019-12-11 12-06-54](https://user-images.githubusercontent.com/12832280/70620404-12940880-1c0f-11ea-9725-e73558e00075.png)

**Comment by justjavac** [comment]
@David-Else I use https://vsmarketplacebadge.apphb.com

**Comment by David-Else** [comment]
@justjavac I don't know about that, I have never made an extension. What happens is when you search for deno in the extensions part of VS Code it finds your extension version `1.4.0`, but as soon as you install it then says `1.3.3`

I don't know where the bug is! I will try again when the new VS Code comes out later this week, maybe that will fix it.

**Comment by David-Else** [comment]
@justjavac Great news, the problem is fixed with the latest VS Code 1.41.0! It now installs vscode-deno 1.4.0 and the extension now works and says:
```
Found deno, version:
deno: 0.26.0
v8: 8.0.192
typescript: 3.7.2
```
I won't close the issue in case there is still a problem for the other people.

**Comment by barthuijgen** [comment]
I can confirm the issue is solved for me too now, thanks! :)

**Comment by rsp** [comment]
@barthuijgen, the default Deno install location for the `deno_install` script has changed (but not for all systems and it doesn't seem to be final - see the comments in denoland/deno_install#89) but people can have it installed somewhere else using the `DENO_INSTALL` env var running the install script, or installing it manually (it's a single file after all, it's easy to drop it into /usr/local/bin if you just want it working with no PATH configuration).

If anayone is interested, here how I am currently installing Deno on CI after the default paths changed:

- https://github.com/rsp/deno-clipboard/commit/b89938ff

In short, it is basically:
```
curl -fsSL https://deno.land/x/install/install.sh \
| sudo DENO_INSTALL=/usr/local sh -s v0.25.0
```
and I could drop the PATH config, as `/usr/local/bin` is already there by default.

**Comment by brandonpsmith** [comment]
Having the same issue. Deno extension cannot find deno.

Mac OSX 10.14.16
VS Code 1.41.1
Deno Extension 1.4.0

Installed deno using asdf as described here https://github.com/denoland/deno_install#install-and-manage-multiple-versions

```bash
which deno
~/.asdf/shims/deno

deno --version
deno 0.30.0
v8 8.1.108
typescript 3.7.2
```

Any ideas?
